src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Orchestration.cs

```diff
@@ -21,105 +20,109 @@ namespace Microsoft.Health.Dicom.Functions.Indexing
+        /// Asynchronously creates an index for the provided query tags over the previously added data.
         /// </summary>
-        /// <param name="context">The context.</param>
-        /// <param name="logger">The logger.</param>
-        /// <returns>The task.</returns>
-        [FunctionName(nameof(ReindexTagsAsync))]
-        public async Task ReindexTagsAsync(
+        /// <remarks>
+        /// Durable functions are reliable, and their implementations will be executed repeatedly over the lifetime of
+        /// a single instance.
+        /// </remarks>
+        /// <param name="context">The context for the orchestration instance.</param>
+        /// <param name="logger">A diagnostic logger.</param>
+        /// <returns>A task representing the <see cref="ReindexInstancesAsync"/> operation.</returns>
+        [FunctionName(nameof(ReindexInstancesAsync))]
+        public async Task ReindexInstancesAsync(
             [OrchestrationTrigger] IDurableOrchestrationContext context,
             ILogger logger)
         {
             EnsureArg.IsNotNull(context, nameof(context));
-            logger = context.CreateReplaySafeLogger(logger);
-            var input = context.GetInput<IReadOnlyCollection<int>>();
-            var tagEntries = await context.CallActivityAsync<IReadOnlyCollection<ExtendedQueryTagStoreEntry>>(nameof(GetTagStoreEntriesAsync), input);
-            var watermarkRange = await context.CallActivityAsync<WatermarkRange>(nameof(GetReindexWatermarkRangeAsync), null);
-            ReindexOperation reindexOperation = new ReindexOperation
-            {
-                OperationId = context.InstanceId,
-                StoreEntries = tagEntries,
-                WatermarkRange = watermarkRange
-            };
-
-            await context.CallSubOrchestratorAsync(nameof(SubReindexTagsAsync), input: reindexOperation);
-        }
 
-        /// <summary>
-        /// The orchestration function to Reindex Extended Query Tags.
-        /// </summary>
-        /// <param name="context">the context.</param>
-        /// <param name="logger">The logger.</param>
-        /// <returns></returns>
-        [FunctionName(nameof(SubReindexTagsAsync))]
-        public async Task SubReindexTagsAsync(
-            [OrchestrationTrigger] IDurableOrchestrationContext context,
-            ILogger logger)
-        {
-            EnsureArg.IsNotNull(context, nameof(context));
             logger = context.CreateReplaySafeLogger(logger);
+            ReindexInput input = context.GetInput<ReindexInput>();
+
+            // Determine the set of query tags that should be indexed and only continue if there is at least 1.
+            // For the first time this orchestration executes, assign all of the tags in the input to the operation,
+            // otherwise simply fetch the tags from the database for this operation.
+            IReadOnlyList<ExtendedQueryTagStoreEntry> queryTags = input.Completed.Count == 0
```

Comment:
I kind of like being consistent with `Count` because if for some reason someone passed the initial progress like [12, 12) that wouldn't contain any data, but it would have `Count == 0`